add overload button

BachelorOogway opened the pull request:
Under Overload mode, user can customize the number of courses they can take each semester, at the same time ensuring not exceeding 12 courses each year.

BachelorOogway committed:
add enable overload button

ShingZhanho commented:
Problem found:
- Even with "overload" enabled, the max. 12 courses limit is not lifted.

Suggestions:
- Maybe the overload button should live in the shopping cart or course selection list only, since the overload option should only be modified during course selection stage. The navigation bar should contain actions related to the entire lifetime of the app.
- The button can be made to be consistent with existing UI elements and UX design. e.g., a button called "Overload Options" that brings up a modal dialogue for modifying the limits?

All these should be rather easy to fix by copilot. **For consistency, we should use Claude Sonnet 4.5 as the model for this repo.**

BachelorOogway committed:
Merge remote-tracking branch 'upstream/master'

BachelorOogway committed:
move overload button to shopping cart, maintain UI design consistency for overload button, enable > 12 courses per academic year

BachelorOogway commented:
Claude Sonnet 4 and Claude Sonnet 4.5 seems to be part of the Pro service of Github Copilot (? T_T
See if this modification by GPT-5-mini lives up to your expectation up till now. It has smaller chat memory but no fault in realising main features given in prompt.

ShingZhanho committed:
refactor: removed pnpm-lock.yaml

This project should use package-lock.json.
Removed pnpm for consistency in dev environments.

ShingZhanho commented:
> Claude Sonnet 4 and Claude Sonnet 4.5 seems to be part of the Pro service of Github Copilot (? T_T See if this modification by GPT-5-mini lives up to your expectation up till now. It has smaller chat memory but no fault in realising main features given in prompt.

They are indeed from Pro service. But you should be able to get it for free by applying the GitHub Education scheme. Anyway, checking it now. Thanks for the work!

ShingZhanho committed:
feat: added pnpm-lock.yaml to .gitignore

ShingZhanho committed:
refactor: UI consistency

- Moved inline CSS into corresponding CSS files.
- Made new components more consistent with existing elements.

ShingZhanho committed:
fix: overload option not being reloaded properly

ShingZhanho committed:
fix: recover old UI and behaviour for shopping cart
